wpilibNewCommands/src/main/native/include/frc2/command/button/CommandJoystick.h

```diff
@@ -15,22 +16,9 @@ namespace frc2 {
  *
  * @see Joystick
  */
-class CommandJoystick : public frc::Joystick {
+class CommandJoystick : public CommandGenericHID, public virtual frc::Joystick {
```

Comment:
Out of curiosity, why was `frc::Joystick` marked as a virtual base class? Are there cases where people would inherit from `CommandJoystick` (or the other command HID classes) and also from `frc::Joystick`?

Reply:
Diamond inheritance is voodoo to me. I think it was so that the `Joystick` superclass uses the `GenericHID` superclass from `CommandGenericHID` rather than having another one.

Reply:
It's voodoo to me too, but looking at the [virtual inheritance docs](https://en.cppreference.com/w/cpp/language/derived_class), I think the shared base class has to be marked final. (see https://godbolt.org/z/Eqaz9n67o)

Reply:
Then it would probably be better for C++ to match Java anyway.